cli/tests/pcluster/config/test_required_params.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from pcluster.config.pcluster_config import PclusterConfig
+
+
+@pytest.mark.parametrize(
+    "cluster_label, expected_message",
+    [
+        ("no_scheduler", "Configuration parameter 'scheduler' must have a value"),
+        ("empty_string_scheduler", "The configuration parameter 'scheduler' has an invalid value"),
+    ],
+)
+def test_param_is_required(mocker, pcluster_config_reader, cluster_label, expected_message):
```

Comment:
If using the "required" attribute we can remove this test module and moving the test of required parameters in the test for the sections. E.g. `test_section_cluster.py` --> `test_cluster_param_from_file`

Reply:
I moved the test cases to the same file, but I created a new function for them because the check happens in the validate method, which is not called by assert_param_from_file (wrapped by test_cluster_param_from_file).